targets/XPlatCppSdk/source/code/source/playfab/PlayFabWinHttpPlugin.cpp

```diff
@@ -32,39 +32,52 @@ namespace PlayFab
 
     void PlayFabWinHttpPlugin::WorkerThread()
     {
-        size_t queueSize;
-
-        while (this->threadRunning)
+        try
         {
-            std::unique_ptr<CallRequestContainerBase> requestContainer = nullptr;
+            size_t queueSize;
 
-            { // LOCK httpRequestMutex
-                std::unique_lock<std::mutex> lock(this->httpRequestMutex);
+            while (this->threadRunning)
+            {
+                std::unique_ptr<CallRequestContainerBase> requestContainer = nullptr;
+
+                { // LOCK httpRequestMutex
+                    std::unique_lock<std::mutex> lock(this->httpRequestMutex);
+
+                    queueSize = this->pendingRequests.size();
+                    if (queueSize != 0)
+                    {
+                        requestContainer = std::move(this->pendingRequests[0]);
+                        this->pendingRequests.pop_front();
+                    }
+                } // UNLOCK httpRequestMutex
 
-                queueSize = this->pendingRequests.size();
-                if (queueSize != 0)
+                if (queueSize == 0)
                 {
-                    requestContainer = std::move(this->pendingRequests[0]);
-                    this->pendingRequests.pop_front();
+                    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+                    continue;
                 }
-            } // UNLOCK httpRequestMutex
 
-            if (queueSize == 0)
-            {
-                std::this_thread::sleep_for(std::chrono::milliseconds(10));
-                continue;
-            }
-
-            if (requestContainer != nullptr)
-            {
-                CallRequestContainer* requestContainerPtr = dynamic_cast<CallRequestContainer*>(requestContainer.get());
-                if (requestContainerPtr != nullptr)
+                if (requestContainer != nullptr)
                 {
-                    requestContainer.release();
-                    ExecuteRequest(std::unique_ptr<CallRequestContainer>(requestContainerPtr));
+                    CallRequestContainer* requestContainerPtr = dynamic_cast<CallRequestContainer*>(requestContainer.get());
+                    if (requestContainerPtr != nullptr)
+                    {
+                        requestContainer.release();
+                        ExecuteRequest(std::unique_ptr<CallRequestContainer>(requestContainerPtr));
+                    }
                 }
             }
         }
+        catch (std::exception ex)
+        {
+            { // LOCK httpRequestMutex
+                std::unique_lock<std::mutex> lock(httpRequestMutex);
+                if (userExceptionCallback)
+                {
+                    userExceptionCallback(ex);
+                }
+            } // UNLOCK httpRequestMutex
```

Comment:
Please call here PlayFabPluginManager.GetInstanceHandlerException(const std::exception&) method instead of all this. Add that method to PlayFabPluginManager. That method will have this mutex block and it will call the user's callback, how you are doing here.